packages/sources/finage-test/src/index.ts

```diff
@@ -1,9 +1,66 @@
 import { expose, ServerInstance } from '@chainlink/external-adapter-framework'
+import { IncludesFile } from '@chainlink/external-adapter-framework/adapter'
 import { PriceAdapter } from '@chainlink/external-adapter-framework/adapter'
 import { config } from './config'
 import { commodities, crypto, eod, forex, stock, ukEtf } from './endpoint'
+import { PriceEndpointInputParametersDefinition } from '@chainlink/external-adapter-framework/adapter/price'
+import { AdapterParams } from '@chainlink/external-adapter-framework/adapter/types'
+import {
+  AdapterRequest,
+  AdapterRequestContext,
+  AdapterResponse,
+} from '@chainlink/external-adapter-framework/util'
 
-export const adapter = new PriceAdapter({
+export type PriceAdapterRequest<T> = AdapterRequest<T> & {
+  requestContext: AdapterRequestContext<T> & {
+    priceMeta: {
+      inverse: boolean
+    }
+  }
+}
+
+class FinageAdapter extends PriceAdapter<any> {
+  constructor(
+    params: AdapterParams<any> & {
+      includes?: IncludesFile
+    },
+  ) {
```

Comment:
Looks like these types could be:

```suggestion
class FinageAdapter<T extends CustomSettingsDefinition> extends PriceAdapter<T> {
  constructor(
    params: AdapterParams<T> & {
      includes?: IncludesFile
    },
  ) {
```